konig-schemagen/src/main/java/io/konig/schemagen/gcp/SpannerTableGenerator.java

```
@@ -0,0 +1,113 @@
+	private OwlReasoner owl;
+	private ShapeNamer shapeNamer;
+	private TableMapper tableMapper;
+	
+	
+	public SpannerTableGenerator(ShapeManager shapeManager) {
+		this.shapeManager = shapeManager;
+	}
+	
+	public SpannerTableGenerator() {
+	}
+
+	public SpannerTableGenerator(ShapeManager shapeManager, ShapeNamer shapeNamer, OwlReasoner reasoner) {
+		this.shapeManager = shapeManager;
+		this.shapeNamer = shapeNamer;
+		owl = reasoner;
+	}
+	
+	public ShapeManager getShapeManager() {
+		return shapeManager;
+	}
+
+	public SpannerTableGenerator setShapeManager(ShapeManager shapeManager) {
+		this.shapeManager = shapeManager;
+		return this;
+	}
+
+	public ShapeNamer getShapeNamer() {
+		return shapeNamer;
+	}
+
+	public SpannerTableGenerator setShapeNamer(ShapeNamer shapeNamer) {
+		this.shapeNamer = shapeNamer;
+		return this;
+	}
+
+	public TableMapper getTableMapper() {
+		return tableMapper;
+	}
+
+
+	public SpannerTableGenerator setTableMapper(TableMapper tableMapper) {
+		this.tableMapper = tableMapper;
+		return this;
+	}
+
+	public void toTableSchema(SpannerTable table) {
+		Shape shape = table.getTableShape();
+		if (shape == null) {
+			throw new SchemaGeneratorException("Shape is not defined");
+		}
+		
+		List<PropertyConstraint> plist = shape.getProperty();
+		
+		for (PropertyConstraint p : plist) {
+			toField(p, table);
+		}
+		
+	}
+	
+
+	private void toField(PropertyConstraint p, SpannerTable table) {
+		
+		String fieldName = p.getPredicate().getLocalName();
+		FieldMode fieldMode = fieldMode(p);
+		SpannerDatatype fieldType = datatypeMap.type(p);
+		
+		SpannerTable.Field field = table.new Field(fieldName, fieldMode, fieldType);
+		table.addField(field);
+		
+		// TBD: To check and add additional info for array type nodes
+	}
+
+	
+	private FieldMode fieldMode(PropertyConstraint p) {
+		Integer minCount = p.getMinCount();
+		Integer maxCount = p.getMaxCount();
+		
+		if (maxCount==null || maxCount>1) {

```

As I said before, it's not clear to me that a REPEATED field is allowed in a Spanner data shape.  Should we throw an exception if a REPEATED field is encountered?